pennylane_lightning/src/gates/cpu_kernels/GateImplementationsPI.hpp

```diff
@@ -89,21 +89,21 @@ class GateImplementationsPI : public PauliGenerator<GateImplementationsPI> {
         GateOperation::MultiRZ};
 
     constexpr static std::array implemented_generators = {
+        GeneratorOperation::PhaseShift,
         GeneratorOperation::RX,
         GeneratorOperation::RY,
         GeneratorOperation::RZ,
-        GeneratorOperation::PhaseShift,
         GeneratorOperation::IsingXX,
         GeneratorOperation::IsingXY,
         GeneratorOperation::IsingYY,
         GeneratorOperation::IsingZZ,
         GeneratorOperation::CRX,
         GeneratorOperation::CRY,
         GeneratorOperation::CRZ,
+        GeneratorOperation::ControlledPhaseShift,
         GeneratorOperation::DoubleExcitation,
```

Comment:
Do we also have the single excitation gates for here?

Reply:
We only have implementations of those gates in the LM kernel. We can add these implementations (and check which one is faster).